raami/queries.js

```diff
@@ -0,0 +1,269 @@
+// var promise = require("bluebird");
+
+// var options = {
+//   // Initialization Options
+//   promiseLib: promise
+// };
+
+// var pgp = require("pg-promise")(options);
+// var db = pgp(process.env.DATABASE_URL)
+
+
+module.exports = {
+  getPeople,
+	getArtists,
+	createArtist,
+	getArtist,
+	updateArtist,
+	getWork,
+	getWorks,
+	createWork,
+	updateWork,
+	removeWork,
+};
+
+function getPeople(req, res, next) {
+  var _id = (req.params.id);                             
+  req.app.locals.db.any(`
+    SELECT id, person_id, name, continent, url, filename, filedata, description,
+           transport, legal, auction, print, digital, agent, contact, waitlist, postage
+      FROM Artist
+     WHERE person_id = $1`, _id
+  )
+    .then(function (data) {
+      res.status(200)
+        .json(data);
+    })
+    .catch(function (err) {
+      return next(err);
+    });
+}
+
+function getArtists(req, res, next) {
+  req.app.locals.db.any('SELECT person_id, id, name FROM Artist')
+    .then(function (data) {
+      res.status(200)
+        .json(data);
+    })
+    .catch(function (err) {
+      return next(err);
+    });
+}
+
+function getArtist(req, res, next) {
+  var _id = (req.params.id);
+  req.app.locals.db.one(`
+    SELECT *
+      FROM Artist
+     WHERE id = $1`, _id
+  )
+    .then(function (data) {
+      res.status(200)
+        .json(data);
+    })
+    .catch(function (err) {
+      return next(err);
+    });
+}
+
+
+function createArtist(req, res, next) {
+
+  req.app.locals.db.one(`
+     INSERT INTO Artist
+                 (
+                   person_id, name, continent, url, filename, filedata,
+                   description, transport, legal, auction, print, digital, agent, contact, waitlist, postage
+                 )
+          VALUES (
+                   $(person_id), $(name), $(continent), $(url), $(filename),
+                   $(filedata), $(description), $(transport), $(legal),
+                   ($(auction)), ($(print)), $(digital), $(agent), $(contact), $(waitlist), ($(postage))
+                 )
+       RETURNING id`, req.body
+  )
```

Comment:
Should require `raami_admin` rights, or check that the email address of `person_id` matches `req.session.user.email`.